### PR TITLE
Added recipe for ZODB

### PR DESCRIPTION
**Pull request by pmlandwehr**
Finally added the recipe for the **Z**ope **O**bject **D**ata**B**ase. Surprised we didn't yet have this.

### CODE REVIEW

**Comment by pmlandwehr** [comment]
Missing:
- [ ] `transaction >=2.0.3`

File: recipes/transaction/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  preserve_egg_dir: True
+  number: {{ build }}
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - zope.interface
```
Comment (synapticarbors) [inline recipes/transaction/meta.yaml]: Does the `build` require `zope.interface`?

File: recipes/transaction/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.{{ bundle }}
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
+  {{ hash_type }}: {{ hash }}
+
+build:
+  preserve_egg_dir: True
```
Comment (synapticarbors) [inline recipes/transaction/meta.yaml]: Does `preserve_egg_dir` have to be set to true? The conda meta.yaml documentation is vague:

https://conda.io/docs/building/meta-yaml.html#preserve-python-egg-directory

so this is an honest question that I'm not sure the answer of.

File: recipes/transaction/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  run:
+    - python
+    - setuptools
+    - zope.interface
+
+test:
+  imports:
+    - transaction
+    - transaction.tests
```
Comment (synapticarbors) [inline recipes/transaction/meta.yaml]: I don't think you need to import the `transaction.tests` module.

File: recipes/zodb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+test:
+  imports:
+    - ZODB
+    - ZODB.FileStorage
+    - ZODB.scripts
+    - ZODB.scripts.tests
+    - ZODB.tests
+
+  commands:
+    # - fsdump --help  # no support for --help
```
Comment (synapticarbors) [inline recipes/zodb/meta.yaml]: I'm not sure if it's necessary, but I guess for the cli's that re missing a help flag, you could check for the existence of them:

http://conda-forge.github.io/docs/testing.html#simple-existence-tests

but I think that's up to you.

File: recipes/transaction/meta.yaml (hunk 2 above)
Comment (pmlandwehr) [inline recipes/transaction/meta.yaml]: `preserve_egg_dir` was included by `conda skeleton`, so I'm inclined to keep it. But yeah, I have no bigger theoretical basis for the choice.

File: recipes/zodb/meta.yaml (hunk 4 above)
Comment (pmlandwehr) [inline recipes/zodb/meta.yaml]: I'm ok leaving these guys as-is for now, I think.

**Pull request merged**